### PR TITLE
315584_configure_embedconfig.json_in_react_with_nodejs_sample

### PR DESCRIPTION
**Pull request by dharshinisf21**

### CODE REVIEW

**Review by vinothkumar1732 — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,79 @@
...
+  * Visual studio code(https://code.visualstudio.com/download)
+
+ ### Run a Sample Using Visual Studio Code
+ 
+  * Open the React with NodeJS sample in Visual Studio Code.
+   
+  * Open the terminal in Visual Studio Code and install all dependent packages by executing the following command in the terminal
+    
+    ```bash
+    npm install
```
[the remark this answers is not in the document]
Comment (dharshinisf21) [reply README.md]: updated space in the comment

**Review by vinothkumar1732 — COMMENTED** [review]
Changes are fine.

File: client/public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <!--
+      This HTML file is a template.
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
```
Comment (Chrispineagunja) [inline client/public/index.html:39]: Run `npm start` or `yarn start` to begin the development.

File: client/public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      To create a production bundle, use `npm run build` or `yarn build`.
```
Comment (Chrispineagunja) [inline client/public/index.html:40]: To create a production bundle, use `npm run build` or `yarn build.`

File: client/src/Dashboard/Dashboard.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import '../index';
+import {BoldBI} from '@boldbi/boldbi-embedded-sdk';
+
+
+//NodeJs application would be run on http://localhost:8080, which needs to be set as `apiHost`
```
Comment (Chrispineagunja) [inline client/src/Dashboard/Dashboard.js:6]: //NodeJs application will run on http://localhost:8080, which needs to be set as `apiHost.`

File: client/src/Dashboard/Dashboard.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+const apiHost="http://localhost:8080";
+
+//Url of the authorizationserver action in ValuesController of the NodeJs application
```
Comment (Chrispineagunja) [inline client/src/Dashboard/Dashboard.js:9]: //Url of the authorizationserver action in the ValuesController of the NodeJs application.

File: embed.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+var http = require("http");
+var https = require("https");
+var url = require("url");
+var express = require('express');
+var cors = require('cors');
+var app = express();
+var crypto = require('crypto');
+const path = require('path');
+app.use(cors());
+// Parse JSON bodies (as sent by API clients)
```
Comment (Chrispineagunja) [inline embed.js:11]: //Parse JSON bodies (as sent by API clients).

File: embed.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+app.use(express.json());
+
+//assign a port number for an API to run,
```
Comment (Chrispineagunja) [inline embed.js:14]: //Assign a port number for an API to run.

File: embed.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+const port = 8080;
+
+let appconfig;
+try {
+  appconfig = JSON.parse(fs.readFileSync('embedConfig.json'));
+} catch (error) {
+  console.error('Error: embedConfig.json file not found.');
+  process.exit(1); // Exit the program with a non-zero exit code to indicate an error
```
Comment (Chrispineagunja) [inline embed.js:22]: process.exit(1); //Exit the program with a non-zero exit code to indicate an error.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,79 @@
...
+   ![Dashboard view](https://github.com/bold-bi/embedded-bi-samples/assets/129486688/36bb7163-1f21-4f76-b347-c9e43fb81b1d)
+ 
+ ## Requirements to run the demo
+
+The samples require the following to run.
+
+ * [NodeJS](https://nodejs.org/en/)
+ * [Visual Studio Code](https://code.visualstudio.com/download)
+
+> **NOTE:** NodeJS v14.16 to v18.16 are supported
```
[the remark this answers is not in the document]
Comment (dharshinisf21) [reply README.md]: updated this line

File: README.md [hunk 10]
```diff
@@ -1,2 +1,79 @@
...
+
+ ## Developer IDE
+
+  * Visual studio code(https://code.visualstudio.com/download)
+
+ ### Run a Sample Using Visual Studio Code
+ 
+  * Open the React with NodeJS sample in **Visual Studio Code**.
+   
+  * Open the terminal in Visual Studio Code and install all dependent packages by executing the following command in the terminal
```
[the remark this answers is not in the document]
Comment (dharshinisf21) [reply README.md]: updated this line

File: README.md [hunk 11]
```diff
@@ -1,2 +1,79 @@
...
+    
+    ```bash
+       npm start
+    ```
+    
+  * After executing the command, the application will automatically launch in the default browser. You can access it at the specified port number (e.g., https://localhost:3000).
+    
+    ![Dashboard view](https://github.com/bold-bi/embedded-bi-samples/assets/129486688/36bb7163-1f21-4f76-b347-c9e43fb81b1d)
+
+> **NOTE:** If the API host is already in use, you can modify the port number as per your preference in embed.js and update that Dashboard.js file.
```
[the remark this answers is not in the document]
Comment (dharshinisf21) [reply README.md]: updated this line

File: README.md [hunk 12]
```diff
@@ -1,2 +1,79 @@
...
+
+Please refer to the [help documentation](https://help.boldbi.com/embedding-options/embedding-sdk/samples/react-with-node-js/#how-to-run-the-sample) to know how to run the sample.
+
+## Important notes
+
+It is recommended to not store passwords and sensitive information in configuration files for security reasons, in a real-world application. Instead, you should consider using a secure application, such as Key Vault, to safeguard your credentials.
```
[the remark this answers is not in the document]
Comment (dharshinisf21) [reply README.md]: updated this line

File: client/public/index.html (hunk 2 above)
Comment (dharshinisf21) [reply client/public/index.html]: updated this line

File: client/public/index.html (hunk 3 above)
Comment (dharshinisf21) [reply client/public/index.html]: updated this line

File: client/src/Dashboard/Dashboard.js (hunk 4 above)
Comment (dharshinisf21) [reply client/src/Dashboard/Dashboard.js]: updated this line

File: client/src/Dashboard/Dashboard.js (hunk 5 above)
Comment (dharshinisf21) [reply client/src/Dashboard/Dashboard.js]: updated this line

File: embed.js (hunk 6 above)
Comment (dharshinisf21) [reply embed.js]: updated this line

File: embed.js (hunk 7 above)
Comment (dharshinisf21) [reply embed.js]: updated this line

File: embed.js (hunk 8 above)
Comment (dharshinisf21) [reply embed.js]: updated this line

File: README.md [hunk 13]
```diff
@@ -1,2 +1,79 @@
...
-This sample demonstrates the dashboard rendering in your Bold BI server in React with NodeJS
+# BoldBI Embedding React with NodeJS Sample
+
+This project was created using NodeJS 14.16. This application aims to demonstrate how to render the dashboard available on your Bold BI server.
+
+## Dashboard view
+
+   ![Dashboard view](https://github.com/bold-bi/embedded-bi-samples/assets/129486688/36bb7163-1f21-4f76-b347-c9e43fb81b1d)
+ 
+ ## Requirements to run the demo
```
Comment (vinothkumar1732) [inline README.md:9]: Requirements/Prerequisites

**Review by vinothkumar1732 — APPROVED** [review]
Sample and readme file are fine.

File: README.md [hunk 14]
```diff
@@ -1,2 +1,79 @@
-# react-with-nodejs-sample
-This sample demonstrates the dashboard rendering in your Bold BI server in React with NodeJS
+# BoldBI Embedding React with NodeJS Sample
+
+This project was created using NodeJS 14.16. This application aims to demonstrate how to render the dashboard available on your Bold BI server.
```
Comment (Chrispineagunja) [inline README.md:3]: This project is created using NodeJS 14.16. This application aims to demonstrate how to render the dashboard available on your Bold BI server.

File: README.md [hunk 15]
```diff
@@ -1,2 +1,79 @@
...
+
+ * https://help.boldbi.com/embedded-bi/faq/where-can-i-find-the-product-version/
+
+ #### Supported browsers
+  
+  * Google Chrome, Microsoft Edge, Mozilla Firefox, and Safari.
+
+ ## Configuration
+
+ * Please ensure that you have enabled embed authentication on the `embed settings` page. If it is not currently enabled, please refer to the following image or detailed [instructions](https://help.boldbi.com/site-administration/embed-settings/#get-embed-secret-code) to enable it.
```
Comment (Chrispineagunja) [inline README.md:28]: * Please enable embed authentication on the `embed settings` page. If it is not currently enabled, please refer to the following image or detailed [instructions](https://help.boldbi.com/site-administration/embed-settings/#get-embed-secret-code) to enable it.

File: README.md [hunk 16]
```diff
@@ -1,2 +1,79 @@
...
+
+    ![Embed Settings](https://github.com/boldbi/aspnet-core-sample/assets/91586758/b3a81978-9eb4-42b2-92bb-d1e2735ab007)
+
+ * To download the `embedConfig.json` file, please follow this [link](https://help.boldbi.com/site-administration/embed-settings/#get-embed-configuration-file) for reference. Additionally, you can refer to the following image for visual guidance.
```
Comment (Chrispineagunja) [inline README.md:32]: * To download the `embedConfig.json` file, please follow this [link](https://help.boldbi.com/site-administration/embed-settings/#get-embed-configuration-file) for reference. Additionally, refer to the following image for visual guidance.

File: README.md [hunk 17]
```diff
@@ -1,2 +1,79 @@
...
+
+    ![Embed Settings Download](https://github.com/boldbi/aspnet-core-sample/assets/91586758/d27d4cfc-6a3e-4c34-975e-f5f22dea6172)
+    ![EmbedConfig Properties](https://github.com/boldbi/aspnet-core-sample/assets/91586758/d6ce925a-0d4c-45d2-817e-24d6d59e0d63)
+
+ * Copy the downloaded `embedConfig.json` file and paste it into the designated [location](https://github.com/boldbi/react-with-nodejs-sample/tree/master) within the application. Please ensure that you have placed it in the application as shown in the following image.
```
Comment (Chrispineagunja) [inline README.md:37]: * Copy the downloaded `embedConfig.json` file and paste it into the designated [location](https://github.com/boldbi/react-with-nodejs-sample/tree/master) within the application. Please ensure you have placed it in the application as shown in the following image.

File: README.md [hunk 18]
```diff
@@ -1,2 +1,79 @@
...
+       npm install
+    ```
+    
+  * Finally, run the application using the following command.
+    
+    ```bash
+       npm start
+    ```
+    
+  * After executing the command, the application will automatically launch in the default browser. You can access it at the specified port number (e.g., https://localhost:3000).
```
Comment (Chrispineagunja) [inline README.md:61]: * The application will automatically launch in the default browser after executing the command. Access it at the specified port number (e.g., https://localhost:3000).

File: README.md [hunk 19]
```diff
@@ -1,2 +1,79 @@
...
+    
+    ![Dashboard view](https://github.com/bold-bi/embedded-bi-samples/assets/129486688/36bb7163-1f21-4f76-b347-c9e43fb81b1d)
+
+> **NOTE:** If the API host is already in use, modify the port number per your preference in embed.js and update that Dashboard.js file.
+
+Please refer to the [help documentation](https://help.boldbi.com/embedding-options/embedding-sdk/samples/react-with-node-js/#how-to-run-the-sample) to know how to run the sample.
+
+## Important notes
+
+In a real-world application, it is recommended not to store passwords and sensitive information in configuration files for security reasons. Instead, you should consider using a secure application, such as Key Vault, to safeguard your credentials.
```
Comment (Chrispineagunja) [inline README.md:71]: In a real-world application, it is recommended not to store passwords and sensitive information in configuration files for security reasons. Instead, you should consider using a secure application like Key Vault to safeguard your credentials.

File: client/public/index.html [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+    <title>ReactWithNodeJS</title>
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <!--
+      This HTML file is a template.
+      If you open it directly in the browser, you will see an empty page.
+
+      You can add webfonts, meta tags, or analytics to this file.
```
Comment (Chrispineagunja) [inline client/public/index.html:36]: Add webfonts, meta tags, or analytics to this file.

**Review by Chrispineagunja — COMMENTED** [review]
Content review completed.

**Pull request merged**